update spec, get proper leader

battlemidget opened the pull request:
First thing is this supports an addition to our spec to allow
a bundle author to create a 00_deploy-done.sh script to test whether
their services are up and can be acted upon.

Second item uses a new shell function in conjure-up to print out a
current leader in a group of units.

For reference: https://github.com/ubuntu/conjure-up/blob/master/share/hooklib/common.sh#L62-L75

Signed-off-by: Adam Stokes <adam.stokes@ubuntu.com>

chuckbutler commented on conjure/steps/00_deploy-done.sh:
```diff
@@ -0,0 +1,14 @@
+
+
+services=("etcd" \
+              "elasticsearch" \
```
nit: indentation here is a little weird

chuckbutler commented on conjure/steps/01_get-kubectl.sh:
```diff
@@ -1,26 +1,15 @@
+juju scp $(getLeader kubernetes):kubectl_package.tar.gz $KUBECTL_PATH/.
 cd $KUBECTL_PATH && tar zxf kubectl_package.tar.gz
 
 exposeResult "Cluster can now be accessed with $KUBECTL_PATH/kubectl application" 0 "true"
```
This is actually pretty cool!

mbruzek commented on conjure/steps/01_get-kubectl.sh:
```diff
@@ -1,26 +1,15 @@
 # TODO: Convert to an actionable item
-juju scp kubernetes/0:kubectl_package.tar.gz $KUBECTL_PATH/.
+debug "Found leader: $(getLeader kubernetes)"
+juju scp $(getLeader kubernetes):kubectl_package.tar.gz $KUBECTL_PATH/.
```
I don't see the getLeader function in this pull request but I imagine it is doing a `juju run service is_leader` which is a time expensive operation. I would recommend saving the leader to a variable and using that instead of calling getLeader twice.

battlemidget replied:
> I don't see the getLeader function in this pull request but I imagine it is doing a juju run service is_leader which is a time expensive operation. I would recommend saving the leader to a variable and using that instead of calling getLeader twice.

Thanks I'll get this fixed up

mbruzek commented:
I don't see how the getLeader function in common.sh would get into this bundle? Is the conjure-up stuff cloned before run or is it just a placeholder?

battlemidget commented:
> I don't see how the getLeader function in common.sh would get into this bundle? Is the conjure-up stuff cloned before run or is it just a placeholder?

The getLeader function is part of the conjure-up package and will always be available prior to deploying this bundle via conjure-up.

battlemidget commented:
@chuckbutler @mbruzek I think i've addressed all your comments.

@mbruzek for reference here is the shell function used: https://github.com/ubuntu/conjure-up/blob/master/share/hooklib/common.sh#L55-L75

mbruzek commented:
+1 LGTM
